From 6371eddd9c0f0a5de2b7e9d46d3de05897c9667d Mon Sep 17 00:00:00 2001
From: napari-bot <81196843+napari-bot@users.noreply.github.com>
Date: Sat, 29 Jul 2023 03:49:26 +0200
Subject: [PATCH] test: [Automatic] Constraints upgrades: `app-model`,
 `certifi`, `dask`, `hypothesis`, `jsonschema`, `npe2`, `pydantic`, `pyqt6`,
 `pyyaml`, `tifffile`, `virtualenv`, `xarray`, `zarr` (#6007)

This PR is automatically created and updated by this napari GitHub
Action:

https://github.com/napari/napari/tree/main/.github/workflows/upgrade_test_contraints.yml
It ensures that dependencies are tested with the most recent version.

The updated packages are:

 * app-model
 * certifi
 * click
 * cmake
 * dask
 * distlib
 * fonttools
 * hypothesis
 * ipykernel
 * jsonschema
 * jsonschema-specifications
 * npe2
 * pip
 * platformdirs
 * pydantic
 * pyqt5-sip
 * pyqt6
 * pyqt6-qt6
 * pyqt6-sip
 * pyyaml
 * referencing
 * rpds-py
 * tifffile
 * tomli-w
 * urllib3
 * virtualenv
 * wheel
 * xarray
 * zarr
 * zipp

---------

Co-authored-by: napari-bot <napari-bot@users.noreply.github.com>
Co-authored-by: Grzegorz Bokota <bokota+github@gmail.com>
---
 napari/plugins/_tests/test_npe2.py            |  7 +-
 napari/types.py                               |  8 +--
 resources/constraints/constraints_py3.10.txt  | 65 +++++++++---------
 resources/constraints/constraints_py3.11.txt  | 64 +++++++++---------
 resources/constraints/constraints_py3.8.txt   | 61 ++++++++---------
 resources/constraints/constraints_py3.9.txt   | 65 +++++++++---------
 .../constraints_py3.9_examples.txt            | 67 ++++++++++---------
 resources/constraints/version_denylist.txt    |  2 +-
 resources/requirements_mypy.txt               | 22 +++---
 setup.cfg                                     |  9 +--
 10 files changed, 186 insertions(+), 184 deletions(-)

diff --git a/napari/plugins/_tests/test_npe2.py b/napari/plugins/_tests/test_npe2.py
index f12ae798..156a57a6 100644
--- a/napari/plugins/_tests/test_npe2.py
+++ b/napari/plugins/_tests/test_npe2.py
@@ -37,9 +37,9 @@ def test_read(mock_pm: 'TestPluginManager'):
     mock_pm.commands.get.reset_mock()
     _, hookimpl = _npe2.read(["some.fzzy"], stack=True)
     mock_pm.commands.get.assert_called_once_with(f'{PLUGIN_NAME}.some_reader')
-
     mock_pm.commands.get.reset_mock()
-    assert _npe2.read(["some.randomext"], stack=True) is None
+    with pytest.raises(ValueError):
+        _npe2.read(["some.randomext"], stack=False)
     mock_pm.commands.get.assert_not_called()
 
     mock_pm.commands.get.reset_mock()
@@ -62,8 +62,7 @@ def test_read(mock_pm: 'TestPluginManager'):
     reason='Older versions of npe2 do not throw specific error.',
 )
 def test_read_with_plugin_failure(mock_pm: 'TestPluginManager'):
-    match = f"Plugin '{PLUGIN_NAME}' was selected"
-    with pytest.raises(ValueError, match=match):
+    with pytest.raises(ValueError):
         _npe2.read(["some.randomext"], stack=True, plugin=PLUGIN_NAME)
 
 
diff --git a/napari/types.py b/napari/types.py
index dd8ad859..6794813a 100644
--- a/napari/types.py
+++ b/napari/types.py
@@ -8,7 +8,6 @@ from typing import (
     Dict,
     Iterable,
     List,
-    Literal,
     NewType,
     Optional,
     Sequence,
@@ -18,6 +17,9 @@ from typing import (
 )
 
 import numpy as np
+
+# TODO decide where types should be defined to have single place for them
+from npe2.types import LayerName as LayerTypeName
 from typing_extensions import TypedDict, get_args
 
 if TYPE_CHECKING:
@@ -64,10 +66,6 @@ __all__ = [
 # since it includes all valid arguments for np.array() ( int, float, str...)
 ArrayLike = Union[np.ndarray, 'dask.array.Array', 'zarr.Array']
 
-LayerTypeName = Literal[
-    "image", "labels", "points", "shapes", "surface", "tracks", "vectors"
-]
-
 # layer data may be: (data,) (data, meta), or (data, meta, layer_type)
 # using "Any" for the data type until ArrayLike is more mature.
 FullLayerData = Tuple[Any, Dict, LayerTypeName]
diff --git a/resources/constraints/constraints_py3.10.txt b/resources/constraints/constraints_py3.10.txt
index a081ce95..405684b7 100644
--- a/resources/constraints/constraints_py3.10.txt
+++ b/resources/constraints/constraints_py3.10.txt
@@ -2,11 +2,11 @@
 # This file is autogenerated by pip-compile with Python 3.10
 # by the following command:
 #
-#    pip-compile --allow-unsafe --extra=pyqt5 --extra=pyqt6_experimental --extra=pyside2 --extra=pyside6_experimental --extra=testing --output-file=resources/constraints/constraints_py3.10.txt --resolver=backtracking --strip-extras resources/constraints/version_denylist.txt setup.cfg
+#    pip-compile --allow-unsafe --extra=pyqt5 --extra=pyqt6_experimental --extra=pyside2 --extra=pyside6_experimental --extra=testing --output-file=resources/constraints/constraints_py3.10.txt --strip-extras resources/constraints/version_denylist.txt setup.cfg
 #
 alabaster==0.7.13
     # via sphinx
-app-model==0.1.4
+app-model==0.2.0
     # via napari (setup.cfg)
 appdirs==1.4.4
     # via
@@ -31,19 +31,19 @@ build==0.10.0
     # via npe2
 cachey==0.2.1
     # via napari (setup.cfg)
-certifi==2023.5.7
+certifi==2023.7.22
     # via
     #   napari (setup.cfg)
     #   requests
 charset-normalizer==3.2.0
     # via requests
-click==8.1.4
+click==8.1.6
     # via
     #   dask
     #   typer
 cloudpickle==2.2.1
     # via dask
-cmake==3.26.4
+cmake==3.27.0
     # via triton
 comm==0.1.3
     # via ipykernel
@@ -53,13 +53,13 @@ coverage==7.2.7
     # via pytest-cov
 cycler==0.11.0
     # via matplotlib
-dask==2023.7.0
+dask==2023.7.1
     # via napari (setup.cfg)
 debugpy==1.6.7
     # via ipykernel
 decorator==5.1.1
     # via ipython
-distlib==0.3.6
+distlib==0.3.7
     # via virtualenv
 docstring-parser==0.15
     # via magicgui
@@ -80,7 +80,7 @@ filelock==3.12.2
     #   torch
     #   triton
     #   virtualenv
-fonttools==4.40.0
+fonttools==4.41.1
     # via matplotlib
 freetype-py==2.4.0
     # via vispy
@@ -92,7 +92,7 @@ heapdict==1.0.1
     # via cachey
 hsluv==5.0.3
     # via vispy
-hypothesis==6.81.0
+hypothesis==6.82.0
     # via napari (setup.cfg)
 idna==3.4
     # via requests
@@ -109,7 +109,7 @@ in-n-out==0.1.8
     # via app-model
 iniconfig==2.0.0
     # via pytest
-ipykernel==6.24.0
+ipykernel==6.25.0
     # via
     #   napari-console
     #   qtconsole
@@ -127,9 +127,9 @@ jinja2==3.1.2
     #   numpydoc
     #   sphinx
     #   torch
-jsonschema==4.18.0
+jsonschema==4.18.4
     # via napari (setup.cfg)
-jsonschema-specifications==2023.6.1
+jsonschema-specifications==2023.7.1
     # via jsonschema
 jupyter-client==8.3.0
     # via
@@ -189,7 +189,7 @@ networkx==3.1
     # via
     #   scikit-image
     #   torch
-npe2==0.7.0
+npe2==0.7.2
     # via
     #   napari (setup.cfg)
     #   napari-plugin-manager
@@ -275,7 +275,7 @@ pillow==10.0.0
     #   scikit-image
 pint==0.22
     # via napari (setup.cfg)
-platformdirs==3.8.1
+platformdirs==3.9.1
     # via
     #   jupyter-core
     #   pooch
@@ -302,7 +302,7 @@ ptyprocess==0.7.0
     # via pexpect
 pure-eval==0.2.2
     # via stack-data
-pydantic==1.10.11
+pydantic==1.10.12
     # via
     #   app-model
     #   napari (setup.cfg)
@@ -327,13 +327,13 @@ pyqt5==5.15.9
     # via napari (setup.cfg)
 pyqt5-qt5==5.15.2
     # via pyqt5
-pyqt5-sip==12.12.1
+pyqt5-sip==12.12.2
     # via pyqt5
-pyqt6==6.5.1
+pyqt6==6.5.2
     # via napari (setup.cfg)
-pyqt6-qt6==6.5.1
+pyqt6-qt6==6.5.2
     # via pyqt6
-pyqt6-sip==13.5.1
+pyqt6-sip==13.5.2
     # via pyqt6
 pyrsistent==0.19.3
     # via jsonschema
@@ -374,13 +374,11 @@ python-dateutil==2.8.2
     #   jupyter-client
     #   matplotlib
     #   pandas
-pytomlpp==1.0.13
-    # via npe2
 pytz==2023.3
     # via pandas
 pywavelets==1.4.1
     # via scikit-image
-pyyaml==6.0
+pyyaml==6.0.1
     # via
     #   dask
     #   napari (setup.cfg)
@@ -402,7 +400,7 @@ qtpy==2.3.1
     #   napari-plugin-manager
     #   qtconsole
     #   superqt
-referencing==0.29.1
+referencing==0.30.0
     # via
     #   jsonschema
     #   jsonschema-specifications
@@ -415,7 +413,7 @@ rich==13.4.2
     #   napari (setup.cfg)
     #   npe2
     #   pytest-pretty
-rpds-py==0.8.10
+rpds-py==0.9.2
     # via
     #   jsonschema
     #   referencing
@@ -469,7 +467,7 @@ tensorstore==0.1.40
     # via
     #   -r resources/constraints/version_denylist.txt
     #   napari (setup.cfg)
-tifffile==2023.7.4
+tifffile==2023.7.18
     # via
     #   napari (setup.cfg)
     #   scikit-image
@@ -477,8 +475,11 @@ tomli==2.0.1
     # via
     #   build
     #   coverage
+    #   npe2
     #   pyproject-hooks
     #   pytest
+tomli-w==1.0.0
+    # via npe2
 toolz==0.12.0
     # via
     #   dask
@@ -520,9 +521,9 @@ typing-extensions==4.7.1
     #   typer
 tzdata==2023.3
     # via pandas
-urllib3==2.0.3
+urllib3==2.0.4
     # via requests
-virtualenv==20.23.1
+virtualenv==20.24.2
     # via napari (setup.cfg)
 vispy==0.13.0
     # via
@@ -530,7 +531,7 @@ vispy==0.13.0
     #   napari-svg
 wcwidth==0.2.6
     # via prompt-toolkit
-wheel==0.40.0
+wheel==0.41.0
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
@@ -540,15 +541,15 @@ wheel==0.40.0
     #   nvidia-nvtx-cu11
 wrapt==1.15.0
     # via napari (setup.cfg)
-xarray==2023.6.0
+xarray==2023.7.0
     # via napari (setup.cfg)
-zarr==2.15.0
+zarr==2.16.0
     # via napari (setup.cfg)
-zipp==3.16.0
+zipp==3.16.2
     # via importlib-metadata
 
 # The following packages are considered to be unsafe in a requirements file:
-pip==23.1.2
+pip==23.2.1
     # via napari-plugin-manager
 setuptools==68.0.0
     # via
diff --git a/resources/constraints/constraints_py3.11.txt b/resources/constraints/constraints_py3.11.txt
index 30092961..28dd119d 100644
--- a/resources/constraints/constraints_py3.11.txt
+++ b/resources/constraints/constraints_py3.11.txt
@@ -2,11 +2,11 @@
 # This file is autogenerated by pip-compile with Python 3.11
 # by the following command:
 #
-#    pip-compile --allow-unsafe --extra=pyqt5 --extra=pyqt6_experimental --extra=pyside2 --extra=pyside6_experimental --extra=testing --output-file=resources/constraints/constraints_py3.11.txt --resolver=backtracking --strip-extras resources/constraints/version_denylist.txt setup.cfg
+#    pip-compile --allow-unsafe --extra=pyqt5 --extra=pyqt6_experimental --extra=pyside2 --extra=pyside6_experimental --extra=testing --output-file=resources/constraints/constraints_py3.11.txt --strip-extras resources/constraints/version_denylist.txt setup.cfg
 #
 alabaster==0.7.13
     # via sphinx
-app-model==0.1.4
+app-model==0.2.0
     # via napari (setup.cfg)
 appdirs==1.4.4
     # via
@@ -31,19 +31,19 @@ build==0.10.0
     # via npe2
 cachey==0.2.1
     # via napari (setup.cfg)
-certifi==2023.5.7
+certifi==2023.7.22
     # via
     #   napari (setup.cfg)
     #   requests
 charset-normalizer==3.2.0
     # via requests
-click==8.1.4
+click==8.1.6
     # via
     #   dask
     #   typer
 cloudpickle==2.2.1
     # via dask
-cmake==3.26.4
+cmake==3.27.0
     # via triton
 comm==0.1.3
     # via ipykernel
@@ -53,13 +53,13 @@ coverage==7.2.7
     # via pytest-cov
 cycler==0.11.0
     # via matplotlib
-dask==2023.7.0
+dask==2023.7.1
     # via napari (setup.cfg)
 debugpy==1.6.7
     # via ipykernel
 decorator==5.1.1
     # via ipython
-distlib==0.3.6
+distlib==0.3.7
     # via virtualenv
 docstring-parser==0.15
     # via magicgui
@@ -76,7 +76,7 @@ filelock==3.12.2
     #   torch
     #   triton
     #   virtualenv
-fonttools==4.40.0
+fonttools==4.41.1
     # via matplotlib
 freetype-py==2.4.0
     # via vispy
@@ -88,7 +88,7 @@ heapdict==1.0.1
     # via cachey
 hsluv==5.0.3
     # via vispy
-hypothesis==6.81.0
+hypothesis==6.82.0
     # via napari (setup.cfg)
 idna==3.4
     # via requests
@@ -105,7 +105,7 @@ in-n-out==0.1.8
     # via app-model
 iniconfig==2.0.0
     # via pytest
-ipykernel==6.24.0
+ipykernel==6.25.0
     # via
     #   napari-console
     #   qtconsole
@@ -123,9 +123,9 @@ jinja2==3.1.2
     #   numpydoc
     #   sphinx
     #   torch
-jsonschema==4.18.0
+jsonschema==4.18.4
     # via napari (setup.cfg)
-jsonschema-specifications==2023.6.1
+jsonschema-specifications==2023.7.1
     # via jsonschema
 jupyter-client==8.3.0
     # via
@@ -185,7 +185,7 @@ networkx==3.1
     # via
     #   scikit-image
     #   torch
-npe2==0.7.0
+npe2==0.7.2
     # via
     #   napari (setup.cfg)
     #   napari-plugin-manager
@@ -271,7 +271,7 @@ pillow==10.0.0
     #   scikit-image
 pint==0.22
     # via napari (setup.cfg)
-platformdirs==3.8.1
+platformdirs==3.9.1
     # via
     #   jupyter-core
     #   pooch
@@ -298,7 +298,7 @@ ptyprocess==0.7.0
     # via pexpect
 pure-eval==0.2.2
     # via stack-data
-pydantic==1.10.11
+pydantic==1.10.12
     # via
     #   app-model
     #   napari (setup.cfg)
@@ -323,13 +323,13 @@ pyqt5==5.15.9
     # via napari (setup.cfg)
 pyqt5-qt5==5.15.2
     # via pyqt5
-pyqt5-sip==12.12.1
+pyqt5-sip==12.12.2
     # via pyqt5
-pyqt6==6.5.1
+pyqt6==6.5.2
     # via napari (setup.cfg)
-pyqt6-qt6==6.5.1
+pyqt6-qt6==6.5.2
     # via pyqt6
-pyqt6-sip==13.5.1
+pyqt6-sip==13.5.2
     # via pyqt6
 pyside2==5.13.2 ; python_version != "3.8"
     # via napari (setup.cfg)
@@ -368,13 +368,11 @@ python-dateutil==2.8.2
     #   jupyter-client
     #   matplotlib
     #   pandas
-pytomlpp==1.0.13
-    # via npe2
 pytz==2023.3
     # via pandas
 pywavelets==1.4.1
     # via scikit-image
-pyyaml==6.0
+pyyaml==6.0.1
     # via
     #   dask
     #   napari (setup.cfg)
@@ -396,7 +394,7 @@ qtpy==2.3.1
     #   napari-plugin-manager
     #   qtconsole
     #   superqt
-referencing==0.29.1
+referencing==0.30.0
     # via
     #   jsonschema
     #   jsonschema-specifications
@@ -409,7 +407,7 @@ rich==13.4.2
     #   napari (setup.cfg)
     #   npe2
     #   pytest-pretty
-rpds-py==0.8.10
+rpds-py==0.9.2
     # via
     #   jsonschema
     #   referencing
@@ -463,10 +461,12 @@ tensorstore==0.1.40
     # via
     #   -r resources/constraints/version_denylist.txt
     #   napari (setup.cfg)
-tifffile==2023.7.4
+tifffile==2023.7.18
     # via
     #   napari (setup.cfg)
     #   scikit-image
+tomli-w==1.0.0
+    # via npe2
 toolz==0.12.0
     # via
     #   dask
@@ -508,9 +508,9 @@ typing-extensions==4.7.1
     #   typer
 tzdata==2023.3
     # via pandas
-urllib3==2.0.3
+urllib3==2.0.4
     # via requests
-virtualenv==20.23.1
+virtualenv==20.24.2
     # via napari (setup.cfg)
 vispy==0.13.0
     # via
@@ -518,7 +518,7 @@ vispy==0.13.0
     #   napari-svg
 wcwidth==0.2.6
     # via prompt-toolkit
-wheel==0.40.0
+wheel==0.41.0
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
@@ -528,15 +528,15 @@ wheel==0.40.0
     #   nvidia-nvtx-cu11
 wrapt==1.15.0
     # via napari (setup.cfg)
-xarray==2023.6.0
+xarray==2023.7.0
     # via napari (setup.cfg)
-zarr==2.15.0
+zarr==2.16.0
     # via napari (setup.cfg)
-zipp==3.16.0
+zipp==3.16.2
     # via importlib-metadata
 
 # The following packages are considered to be unsafe in a requirements file:
-pip==23.1.2
+pip==23.2.1
     # via napari-plugin-manager
 setuptools==68.0.0
     # via
diff --git a/resources/constraints/constraints_py3.8.txt b/resources/constraints/constraints_py3.8.txt
index 4e82451a..861f713c 100644
--- a/resources/constraints/constraints_py3.8.txt
+++ b/resources/constraints/constraints_py3.8.txt
@@ -2,11 +2,11 @@
 # This file is autogenerated by pip-compile with Python 3.8
 # by the following command:
 #
-#    pip-compile --allow-unsafe --extra=pyqt5 --extra=pyqt6_experimental --extra=pyside2 --extra=pyside6_experimental --extra=testing --output-file=resources/constraints/constraints_py3.8.txt --resolver=backtracking --strip-extras resources/constraints/version_denylist.txt setup.cfg
+#    pip-compile --allow-unsafe --extra=pyqt5 --extra=pyqt6_experimental --extra=pyside2 --extra=pyside6_experimental --extra=testing --output-file=resources/constraints/constraints_py3.8.txt --strip-extras resources/constraints/version_denylist.txt setup.cfg
 #
 alabaster==0.7.13
     # via sphinx
-app-model==0.1.4
+app-model==0.2.0
     # via napari (setup.cfg)
 appdirs==1.4.4
     # via
@@ -31,19 +31,19 @@ build==0.10.0
     # via npe2
 cachey==0.2.1
     # via napari (setup.cfg)
-certifi==2023.5.7
+certifi==2023.7.22
     # via
     #   napari (setup.cfg)
     #   requests
 charset-normalizer==3.2.0
     # via requests
-click==8.1.4
+click==8.1.6
     # via
     #   dask
     #   typer
 cloudpickle==2.2.1
     # via dask
-cmake==3.26.4
+cmake==3.27.0
     # via triton
 comm==0.1.3
     # via ipykernel
@@ -59,7 +59,7 @@ debugpy==1.6.7
     # via ipykernel
 decorator==5.1.1
     # via ipython
-distlib==0.3.6
+distlib==0.3.7
     # via virtualenv
 docstring-parser==0.15
     # via magicgui
@@ -80,7 +80,7 @@ filelock==3.12.2
     #   torch
     #   triton
     #   virtualenv
-fonttools==4.40.0
+fonttools==4.41.1
     # via matplotlib
 freetype-py==2.4.0
     # via vispy
@@ -92,7 +92,7 @@ heapdict==1.0.1
     # via cachey
 hsluv==5.0.3
     # via vispy
-hypothesis==6.81.0
+hypothesis==6.82.0
     # via napari (setup.cfg)
 idna==3.4
     # via requests
@@ -117,7 +117,7 @@ in-n-out==0.1.8
     # via app-model
 iniconfig==2.0.0
     # via pytest
-ipykernel==6.24.0
+ipykernel==6.25.0
     # via
     #   napari-console
     #   qtconsole
@@ -135,9 +135,9 @@ jinja2==3.1.2
     #   numpydoc
     #   sphinx
     #   torch
-jsonschema==4.18.0
+jsonschema==4.18.4
     # via napari (setup.cfg)
-jsonschema-specifications==2023.6.1
+jsonschema-specifications==2023.7.1
     # via jsonschema
 jupyter-client==8.3.0
     # via
@@ -197,7 +197,7 @@ networkx==3.1
     # via
     #   scikit-image
     #   torch
-npe2==0.7.0
+npe2==0.7.2
     # via
     #   napari (setup.cfg)
     #   napari-plugin-manager
@@ -285,7 +285,7 @@ pint==0.21.1
     # via napari (setup.cfg)
 pkgutil-resolve-name==1.3.10
     # via jsonschema
-platformdirs==3.8.1
+platformdirs==3.9.1
     # via
     #   jupyter-core
     #   pooch
@@ -312,7 +312,7 @@ ptyprocess==0.7.0
     # via pexpect
 pure-eval==0.2.2
     # via stack-data
-pydantic==1.10.11
+pydantic==1.10.12
     # via
     #   app-model
     #   napari (setup.cfg)
@@ -337,13 +337,13 @@ pyqt5==5.15.9
     # via napari (setup.cfg)
 pyqt5-qt5==5.15.2
     # via pyqt5
-pyqt5-sip==12.12.1
+pyqt5-sip==12.12.2
     # via pyqt5
-pyqt6==6.5.1
+pyqt6==6.5.2
     # via napari (setup.cfg)
-pyqt6-qt6==6.5.1
+pyqt6-qt6==6.5.2
     # via pyqt6
-pyqt6-sip==13.5.1
+pyqt6-sip==13.5.2
     # via pyqt6
 pyside2==5.15.2.1 ; python_version == "3.8"
     # via napari (setup.cfg)
@@ -382,15 +382,13 @@ python-dateutil==2.8.2
     #   jupyter-client
     #   matplotlib
     #   pandas
-pytomlpp==1.0.13
-    # via npe2
 pytz==2023.3
     # via
     #   babel
     #   pandas
 pywavelets==1.4.1
     # via scikit-image
-pyyaml==6.0
+pyyaml==6.0.1
     # via
     #   dask
     #   napari (setup.cfg)
@@ -412,7 +410,7 @@ qtpy==2.3.1
     #   napari-plugin-manager
     #   qtconsole
     #   superqt
-referencing==0.29.1
+referencing==0.30.0
     # via
     #   jsonschema
     #   jsonschema-specifications
@@ -425,7 +423,7 @@ rich==13.4.2
     #   napari (setup.cfg)
     #   npe2
     #   pytest-pretty
-rpds-py==0.8.10
+rpds-py==0.9.2
     # via
     #   jsonschema
     #   referencing
@@ -479,7 +477,7 @@ tensorstore==0.1.40
     # via
     #   -r resources/constraints/version_denylist.txt
     #   napari (setup.cfg)
-tifffile==2023.7.4
+tifffile==2023.7.10
     # via
     #   napari (setup.cfg)
     #   scikit-image
@@ -487,8 +485,11 @@ tomli==2.0.1
     # via
     #   build
     #   coverage
+    #   npe2
     #   pyproject-hooks
     #   pytest
+tomli-w==1.0.0
+    # via npe2
 toolz==0.12.0
     # via
     #   dask
@@ -531,9 +532,9 @@ typing-extensions==4.7.1
     #   typer
 tzdata==2023.3
     # via pandas
-urllib3==2.0.3
+urllib3==2.0.4
     # via requests
-virtualenv==20.23.1
+virtualenv==20.24.2
     # via napari (setup.cfg)
 vispy==0.13.0
     # via
@@ -541,7 +542,7 @@ vispy==0.13.0
     #   napari-svg
 wcwidth==0.2.6
     # via prompt-toolkit
-wheel==0.40.0
+wheel==0.41.0
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
@@ -553,15 +554,15 @@ wrapt==1.15.0
     # via napari (setup.cfg)
 xarray==2023.1.0
     # via napari (setup.cfg)
-zarr==2.15.0
+zarr==2.16.0
     # via napari (setup.cfg)
-zipp==3.16.0
+zipp==3.16.2
     # via
     #   importlib-metadata
     #   importlib-resources
 
 # The following packages are considered to be unsafe in a requirements file:
-pip==23.1.2
+pip==23.2.1
     # via napari-plugin-manager
 setuptools==68.0.0
     # via
diff --git a/resources/constraints/constraints_py3.9.txt b/resources/constraints/constraints_py3.9.txt
index 696797fd..35dc364d 100644
--- a/resources/constraints/constraints_py3.9.txt
+++ b/resources/constraints/constraints_py3.9.txt
@@ -2,11 +2,11 @@
 # This file is autogenerated by pip-compile with Python 3.9
 # by the following command:
 #
-#    pip-compile --allow-unsafe --extra=pyqt5 --extra=pyqt6_experimental --extra=pyside2 --extra=pyside6_experimental --extra=testing --output-file=resources/constraints/constraints_py3.9.txt --resolver=backtracking --strip-extras resources/constraints/version_denylist.txt setup.cfg
+#    pip-compile --allow-unsafe --extra=pyqt5 --extra=pyqt6_experimental --extra=pyside2 --extra=pyside6_experimental --extra=testing --output-file=resources/constraints/constraints_py3.9.txt --strip-extras resources/constraints/version_denylist.txt setup.cfg
 #
 alabaster==0.7.13
     # via sphinx
-app-model==0.1.4
+app-model==0.2.0
     # via napari (setup.cfg)
 appdirs==1.4.4
     # via
@@ -31,19 +31,19 @@ build==0.10.0
     # via npe2
 cachey==0.2.1
     # via napari (setup.cfg)
-certifi==2023.5.7
+certifi==2023.7.22
     # via
     #   napari (setup.cfg)
     #   requests
 charset-normalizer==3.2.0
     # via requests
-click==8.1.4
+click==8.1.6
     # via
     #   dask
     #   typer
 cloudpickle==2.2.1
     # via dask
-cmake==3.26.4
+cmake==3.27.0
     # via triton
 comm==0.1.3
     # via ipykernel
@@ -53,13 +53,13 @@ coverage==7.2.7
     # via pytest-cov
 cycler==0.11.0
     # via matplotlib
-dask==2023.7.0
+dask==2023.7.1
     # via napari (setup.cfg)
 debugpy==1.6.7
     # via ipykernel
 decorator==5.1.1
     # via ipython
-distlib==0.3.6
+distlib==0.3.7
     # via virtualenv
 docstring-parser==0.15
     # via magicgui
@@ -80,7 +80,7 @@ filelock==3.12.2
     #   torch
     #   triton
     #   virtualenv
-fonttools==4.40.0
+fonttools==4.41.1
     # via matplotlib
 freetype-py==2.4.0
     # via vispy
@@ -92,7 +92,7 @@ heapdict==1.0.1
     # via cachey
 hsluv==5.0.3
     # via vispy
-hypothesis==6.81.0
+hypothesis==6.82.0
     # via napari (setup.cfg)
 idna==3.4
     # via requests
@@ -114,7 +114,7 @@ in-n-out==0.1.8
     # via app-model
 iniconfig==2.0.0
     # via pytest
-ipykernel==6.24.0
+ipykernel==6.25.0
     # via
     #   napari-console
     #   qtconsole
@@ -132,9 +132,9 @@ jinja2==3.1.2
     #   numpydoc
     #   sphinx
     #   torch
-jsonschema==4.18.0
+jsonschema==4.18.4
     # via napari (setup.cfg)
-jsonschema-specifications==2023.6.1
+jsonschema-specifications==2023.7.1
     # via jsonschema
 jupyter-client==8.3.0
     # via
@@ -194,7 +194,7 @@ networkx==3.1
     # via
     #   scikit-image
     #   torch
-npe2==0.7.0
+npe2==0.7.2
     # via
     #   napari (setup.cfg)
     #   napari-plugin-manager
@@ -280,7 +280,7 @@ pillow==10.0.0
     #   scikit-image
 pint==0.22
     # via napari (setup.cfg)
-platformdirs==3.8.1
+platformdirs==3.9.1
     # via
     #   jupyter-core
     #   pooch
@@ -307,7 +307,7 @@ ptyprocess==0.7.0
     # via pexpect
 pure-eval==0.2.2
     # via stack-data
-pydantic==1.10.11
+pydantic==1.10.12
     # via
     #   app-model
     #   napari (setup.cfg)
@@ -332,13 +332,13 @@ pyqt5==5.15.9
     # via napari (setup.cfg)
 pyqt5-qt5==5.15.2
     # via pyqt5
-pyqt5-sip==12.12.1
+pyqt5-sip==12.12.2
     # via pyqt5
-pyqt6==6.5.1
+pyqt6==6.5.2
     # via napari (setup.cfg)
-pyqt6-qt6==6.5.1
+pyqt6-qt6==6.5.2
     # via pyqt6
-pyqt6-sip==13.5.1
+pyqt6-sip==13.5.2
     # via pyqt6
 pyside2==5.15.2.1 ; python_version != "3.8"
     # via napari (setup.cfg)
@@ -377,13 +377,11 @@ python-dateutil==2.8.2
     #   jupyter-client
     #   matplotlib
     #   pandas
-pytomlpp==1.0.13
-    # via npe2
 pytz==2023.3
     # via pandas
 pywavelets==1.4.1
     # via scikit-image
-pyyaml==6.0
+pyyaml==6.0.1
     # via
     #   dask
     #   napari (setup.cfg)
@@ -405,7 +403,7 @@ qtpy==2.3.1
     #   napari-plugin-manager
     #   qtconsole
     #   superqt
-referencing==0.29.1
+referencing==0.30.0
     # via
     #   jsonschema
     #   jsonschema-specifications
@@ -418,7 +416,7 @@ rich==13.4.2
     #   napari (setup.cfg)
     #   npe2
     #   pytest-pretty
-rpds-py==0.8.10
+rpds-py==0.9.2
     # via
     #   jsonschema
     #   referencing
@@ -472,7 +470,7 @@ tensorstore==0.1.40
     # via
     #   -r resources/constraints/version_denylist.txt
     #   napari (setup.cfg)
-tifffile==2023.7.4
+tifffile==2023.7.18
     # via
     #   napari (setup.cfg)
     #   scikit-image
@@ -480,8 +478,11 @@ tomli==2.0.1
     # via
     #   build
     #   coverage
+    #   npe2
     #   pyproject-hooks
     #   pytest
+tomli-w==1.0.0
+    # via npe2
 toolz==0.12.0
     # via
     #   dask
@@ -524,9 +525,9 @@ typing-extensions==4.7.1
     #   typer
 tzdata==2023.3
     # via pandas
-urllib3==2.0.3
+urllib3==2.0.4
     # via requests
-virtualenv==20.23.1
+virtualenv==20.24.2
     # via napari (setup.cfg)
 vispy==0.13.0
     # via
@@ -534,7 +535,7 @@ vispy==0.13.0
     #   napari-svg
 wcwidth==0.2.6
     # via prompt-toolkit
-wheel==0.40.0
+wheel==0.41.0
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
@@ -544,17 +545,17 @@ wheel==0.40.0
     #   nvidia-nvtx-cu11
 wrapt==1.15.0
     # via napari (setup.cfg)
-xarray==2023.6.0
+xarray==2023.7.0
     # via napari (setup.cfg)
-zarr==2.15.0
+zarr==2.16.0
     # via napari (setup.cfg)
-zipp==3.16.0
+zipp==3.16.2
     # via
     #   importlib-metadata
     #   importlib-resources
 
 # The following packages are considered to be unsafe in a requirements file:
-pip==23.1.2
+pip==23.2.1
     # via napari-plugin-manager
 setuptools==68.0.0
     # via
diff --git a/resources/constraints/constraints_py3.9_examples.txt b/resources/constraints/constraints_py3.9_examples.txt
index 384cdc65..f1483ac2 100644
--- a/resources/constraints/constraints_py3.9_examples.txt
+++ b/resources/constraints/constraints_py3.9_examples.txt
@@ -2,11 +2,11 @@
 # This file is autogenerated by pip-compile with Python 3.9
 # by the following command:
 #
-#    pip-compile --allow-unsafe --extra=pyqt5 --extra=pyqt6_experimental --extra=pyside2 --extra=pyside6_experimental --extra=testing --output-file=resources/constraints/constraints_py3.9_examples.txt --resolver=backtracking --strip-extras resources/constraints/version_denylist.txt resources/constraints/version_denylist_examples.txt setup.cfg
+#    pip-compile --allow-unsafe --extra=pyqt5 --extra=pyqt6_experimental --extra=pyside2 --extra=pyside6_experimental --extra=testing --output-file=resources/constraints/constraints_py3.9_examples.txt --strip-extras resources/constraints/version_denylist.txt resources/constraints/version_denylist_examples.txt setup.cfg
 #
 alabaster==0.7.13
     # via sphinx
-app-model==0.1.4
+app-model==0.2.0
     # via napari (setup.cfg)
 appdirs==1.4.4
     # via
@@ -31,19 +31,19 @@ build==0.10.0
     # via npe2
 cachey==0.2.1
     # via napari (setup.cfg)
-certifi==2023.5.7
+certifi==2023.7.22
     # via
     #   napari (setup.cfg)
     #   requests
 charset-normalizer==3.2.0
     # via requests
-click==8.1.4
+click==8.1.6
     # via
     #   dask
     #   typer
 cloudpickle==2.2.1
     # via dask
-cmake==3.26.4
+cmake==3.27.0
     # via triton
 comm==0.1.3
     # via ipykernel
@@ -53,13 +53,13 @@ coverage==7.2.7
     # via pytest-cov
 cycler==0.11.0
     # via matplotlib
-dask==2023.7.0
+dask==2023.7.1
     # via napari (setup.cfg)
 debugpy==1.6.7
     # via ipykernel
 decorator==5.1.1
     # via ipython
-distlib==0.3.6
+distlib==0.3.7
     # via virtualenv
 docstring-parser==0.15
     # via magicgui
@@ -80,7 +80,7 @@ filelock==3.12.2
     #   torch
     #   triton
     #   virtualenv
-fonttools==4.40.0
+fonttools==4.41.1
     # via matplotlib
 freetype-py==2.4.0
     # via vispy
@@ -92,7 +92,7 @@ heapdict==1.0.1
     # via cachey
 hsluv==5.0.3
     # via vispy
-hypothesis==6.81.0
+hypothesis==6.82.0
     # via napari (setup.cfg)
 idna==3.4
     # via requests
@@ -114,7 +114,7 @@ in-n-out==0.1.8
     # via app-model
 iniconfig==2.0.0
     # via pytest
-ipykernel==6.24.0
+ipykernel==6.25.0
     # via
     #   napari-console
     #   qtconsole
@@ -136,9 +136,9 @@ joblib==1.3.1
     # via
     #   nilearn
     #   scikit-learn
-jsonschema==4.18.0
+jsonschema==4.18.4
     # via napari (setup.cfg)
-jsonschema-specifications==2023.6.1
+jsonschema-specifications==2023.7.1
     # via jsonschema
 jupyter-client==8.3.0
     # via
@@ -204,7 +204,7 @@ nibabel==5.1.0
     # via nilearn
 nilearn==0.10.1
     # via -r resources/constraints/version_denylist_examples.txt
-npe2==0.7.0
+npe2==0.7.2
     # via
     #   napari (setup.cfg)
     #   napari-plugin-manager
@@ -298,7 +298,7 @@ pillow==10.0.0
     #   scikit-image
 pint==0.22
     # via napari (setup.cfg)
-platformdirs==3.8.1
+platformdirs==3.9.1
     # via
     #   jupyter-core
     #   pooch
@@ -325,7 +325,7 @@ ptyprocess==0.7.0
     # via pexpect
 pure-eval==0.2.2
     # via stack-data
-pydantic==1.10.11
+pydantic==1.10.12
     # via
     #   app-model
     #   napari (setup.cfg)
@@ -350,13 +350,13 @@ pyqt5==5.15.9
     # via napari (setup.cfg)
 pyqt5-qt5==5.15.2
     # via pyqt5
-pyqt5-sip==12.12.1
+pyqt5-sip==12.12.2
     # via pyqt5
-pyqt6==6.5.1
+pyqt6==6.5.2
     # via napari (setup.cfg)
-pyqt6-qt6==6.5.1
+pyqt6-qt6==6.5.2
     # via pyqt6
-pyqt6-sip==13.5.1
+pyqt6-sip==13.5.2
     # via pyqt6
 pyside2==5.15.2.1 ; python_version != "3.8"
     # via napari (setup.cfg)
@@ -395,13 +395,11 @@ python-dateutil==2.8.2
     #   jupyter-client
     #   matplotlib
     #   pandas
-pytomlpp==1.0.13
-    # via npe2
 pytz==2023.3
     # via pandas
 pywavelets==1.4.1
     # via scikit-image
-pyyaml==6.0
+pyyaml==6.0.1
     # via
     #   dask
     #   napari (setup.cfg)
@@ -423,7 +421,7 @@ qtpy==2.3.1
     #   napari-plugin-manager
     #   qtconsole
     #   superqt
-referencing==0.29.1
+referencing==0.30.0
     # via
     #   jsonschema
     #   jsonschema-specifications
@@ -437,7 +435,7 @@ rich==13.4.2
     #   napari (setup.cfg)
     #   npe2
     #   pytest-pretty
-rpds-py==0.8.10
+rpds-py==0.9.2
     # via
     #   jsonschema
     #   referencing
@@ -495,9 +493,9 @@ tensorstore==0.1.40
     # via
     #   -r resources/constraints/version_denylist.txt
     #   napari (setup.cfg)
-threadpoolctl==3.1.0
+threadpoolctl==3.2.0
     # via scikit-learn
-tifffile==2023.7.4
+tifffile==2023.7.18
     # via
     #   napari (setup.cfg)
     #   scikit-image
@@ -505,8 +503,11 @@ tomli==2.0.1
     # via
     #   build
     #   coverage
+    #   npe2
     #   pyproject-hooks
     #   pytest
+tomli-w==1.0.0
+    # via npe2
 toolz==0.12.0
     # via
     #   dask
@@ -549,9 +550,9 @@ typing-extensions==4.7.1
     #   typer
 tzdata==2023.3
     # via pandas
-urllib3==2.0.3
+urllib3==2.0.4
     # via requests
-virtualenv==20.23.1
+virtualenv==20.24.2
     # via napari (setup.cfg)
 vispy==0.13.0
     # via
@@ -559,7 +560,7 @@ vispy==0.13.0
     #   napari-svg
 wcwidth==0.2.6
     # via prompt-toolkit
-wheel==0.40.0
+wheel==0.41.0
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
@@ -569,17 +570,17 @@ wheel==0.40.0
     #   nvidia-nvtx-cu11
 wrapt==1.15.0
     # via napari (setup.cfg)
-xarray==2023.6.0
+xarray==2023.7.0
     # via napari (setup.cfg)
-zarr==2.15.0
+zarr==2.16.0
     # via napari (setup.cfg)
-zipp==3.16.0
+zipp==3.16.2
     # via
     #   importlib-metadata
     #   importlib-resources
 
 # The following packages are considered to be unsafe in a requirements file:
-pip==23.1.2
+pip==23.2.1
     # via napari-plugin-manager
 setuptools==68.0.0
     # via
diff --git a/resources/constraints/version_denylist.txt b/resources/constraints/version_denylist.txt
index 629e74f5..049adef2 100644
--- a/resources/constraints/version_denylist.txt
+++ b/resources/constraints/version_denylist.txt
@@ -1,6 +1,6 @@
 pytest-cov
 PySide6 < 6.3.2 ; python_version < '3.10'
-PySide6 != 6.4.3, !=6.5.0, !=6.5.1, !=6.5.1.1; python_version >= '3.10'
+PySide6 != 6.4.3, !=6.5.0, !=6.5.1, !=6.5.1.1, !=6.5.2; python_version >= '3.10'
 pytest-json-report
 pyopengl!=3.1.7
 tensorstore!=0.1.38
diff --git a/resources/requirements_mypy.txt b/resources/requirements_mypy.txt
index 285c1669..b9746cf5 100644
--- a/resources/requirements_mypy.txt
+++ b/resources/requirements_mypy.txt
@@ -2,13 +2,13 @@
 # This file is autogenerated by pip-compile with Python 3.11
 # by the following command:
 #
-#    pip-compile --config=pyproject.toml --output-file=resources/requirements_mypy.txt --resolver=backtracking resources/requirements_mypy.in
+#    pip-compile --config=pyproject.toml --output-file=resources/requirements_mypy.txt resources/requirements_mypy.in
 #
 appdirs==1.4.4
     # via npe2
 build==0.10.0
     # via npe2
-click==8.1.4
+click==8.1.6
     # via typer
 docstring-parser==0.15
     # via magicgui
@@ -24,7 +24,7 @@ mypy-extensions==1.0.0
     # via
     #   mypy
     #   psygnal
-npe2==0.7.0
+npe2==0.7.2
     # via -r resources/requirements_mypy.in
 numpy==1.25.1
     # via -r resources/requirements_mypy.in
@@ -37,7 +37,7 @@ psygnal==0.9.1
     # via
     #   magicgui
     #   npe2
-pydantic==1.10.11
+pydantic==1.10.12
     # via
     #   -r resources/requirements_mypy.in
     #   npe2
@@ -47,9 +47,7 @@ pygments==2.15.1
     #   superqt
 pyproject-hooks==1.0.0
     # via build
-pytomlpp==1.0.13
-    # via npe2
-pyyaml==6.0
+pyyaml==6.0.1
     # via npe2
 qtpy==2.3.1
     # via
@@ -60,15 +58,17 @@ rich==13.4.2
     # via npe2
 superqt==0.4.1
     # via magicgui
+tomli-w==1.0.0
+    # via npe2
 typer==0.9.0
     # via npe2
-types-pyyaml==6.0.12.10
+types-pyyaml==6.0.12.11
     # via -r resources/requirements_mypy.in
-types-requests==2.31.0.1
+types-requests==2.31.0.2
     # via -r resources/requirements_mypy.in
-types-setuptools==68.0.0.1
+types-setuptools==68.0.0.3
     # via -r resources/requirements_mypy.in
-types-urllib3==1.26.25.13
+types-urllib3==1.26.25.14
     # via types-requests
 typing-extensions==4.7.1
     # via
diff --git a/setup.cfg b/setup.cfg
index cd82241d..f37ea199 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -44,7 +44,7 @@ python_requires = >=3.8
 include_package_data = True
 install_requires =
     appdirs>=1.4.4
-    app-model>=0.1.0,<0.3.0  # as per @czaki request. app-model v0.3.0 can drop napari v0.4.17
+    app-model>=0.1.2,<0.3.0  # as per @czaki request. app-model v0.3.0 can drop napari v0.4.17
     cachey>=0.2.1
     certifi>=2018.1.18
     dask[array]>=2.15.0,!=2.28.0  # https://github.com/napari/napari/issues/1656
@@ -54,9 +54,9 @@ install_requires =
     magicgui>=0.3.6
     napari-console>=0.0.6
     napari-plugin-engine>=0.1.9
-    napari-svg>=0.1.7
-    npe2>=0.5.2
-    numpy>=1.21
+    napari-svg>=0.1.8
+    npe2>=0.7.2
+    numpy>=1.21,<2
     numpydoc>=0.9.2
     pandas>=1.1.0 ; python_version < '3.9'
     pandas>=1.3.0 ; python_version >= '3.9'
@@ -134,6 +134,7 @@ testing =
     IPython>=7.25.0
     qtconsole>=4.5.1
     rich>=12.0.0
+    napari-plugin-manager >=0.1.0a2, <0.2.0
 release = 
     PyGithub>=1.44.1
     twine>=3.1.1
-- 
2.34.1
